crates/runtime/src/actor/component/mod.rs

```diff
@@ -489,21 +435,26 @@ impl GuestBindings {
         response: impl AsyncWrite + Send + Sync + Unpin + 'static,
     ) -> anyhow::Result<Result<(), String>> {
         let ctx = store.data_mut();
-        ctx.stdin.replace(Box::new(request)).await;
-        ctx.stdout.replace(Box::new(response)).await;
+        ctx.stdin
+            .replace(Box::new(AsyncReadStream::new(request)))
+            .await;
+        ctx.stdout
+            .replace(Box::new(AsyncWriteStream::new(1 << 16, response)))
+            .await;
         let res = match self {
             GuestBindings::Command(bindings) => {
                 let operation = operation.as_ref();
                 let wasi = preview2::WasiCtxBuilder::new()
-                    .set_args(&["main.wasm", operation]) // TODO: Configure argv[0]
-                    .set_stdin(ctx.stdin.clone())
-                    .set_stdout(ctx.stdout.clone())
-                    .set_stderr(ctx.stderr.clone())
+                    .args(&["main.wasm", operation]) // TODO: Configure argv[0]
+                    .stdin(ctx.stdin.clone(), preview2::IsATTY::No)
+                    .stdout(ctx.stdout.clone(), preview2::IsATTY::No)
+                    .stderr(ctx.stderr.clone(), preview2::IsATTY::No)
                     .build(&mut ctx.table)
                     .context("failed to build WASI")?;
                 let wasi = replace(&mut ctx.wasi, wasi);
                 trace!(operation, "call `wasi:command/command.run`");
                 let res = bindings
+                    .wasi_cli_run()
```

Comment:
Since this is from bindgen I'm struggling to figure out where it's defined/what it does. Can you help?

Reply:
Sure, it's generated from this interface https://github.com/WebAssembly/wasi-cli/blob/4fe2acb65f18120b463dd0c790b6cc31a8eaa07a/wit/run.wit#L1-L4

This basically means "select `wasi:cli/run` interface"